apps/typegpu-docs/src/content/examples/rendering/3d-fish/render.ts

```diff
@@ -0,0 +1,147 @@
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+import * as std from 'typegpu/std';
+import * as p from './params';
+import {
+  ModelVertexInput,
+  ModelVertexOutput,
+  renderBindGroupLayout,
+} from './schemas';
+import { hsvToRgb, rgbToHsv } from './tgsl-helpers';
+
+const { camera, modelTexture, sampler, modelData } =
+  renderBindGroupLayout.bound;
+
+export const vertexShader = tgpu['~unstable']
+  .vertexFn({
+    in: { ...ModelVertexInput, instanceIndex: d.builtin.instanceIndex },
+    out: ModelVertexOutput,
+  })
+  .does((input) => {
+    // rotate the model so that it aligns with model's direction of movement
+    // https://simple.wikipedia.org/wiki/Pitch,_yaw,_and_roll
+    const currentModelData = modelData.value[input.instanceIndex];
+
+    const modelPosition = input.modelPosition;
+
+    const direction = std.normalize(currentModelData.direction);
+
+    const yaw = std.atan2(direction.z, direction.x) + Math.PI;
+    // biome-ignore format:
+    const yawMatrix = d.mat3x3f(
+          std.cos(yaw),  0, std.sin(yaw),
+          0,             1, 0,           
+          -std.sin(yaw), 0, std.cos(yaw),
+        );
+
+    const pitch = -std.asin(-direction.y);
+    // biome-ignore format:
+    const pitchMatrix = d.mat3x3f(
+          std.cos(pitch), -std.sin(pitch), 0,
+          std.sin(pitch), std.cos(pitch),  0,
+          0,              0,               1,
+        );
```

Comment:
Even though the formatting is a bit different from the default, I would apply the same indentation levels.
```suggestion
    const pitchMatrix = d.mat3x3f(
      std.cos(pitch), -std.sin(pitch), 0,
      std.sin(pitch), std.cos(pitch),  0,
      0,              0,               1,
    );
```